local/core/client.go

```diff
@@ -29,12 +29,12 @@ func checkTLSConfig(cfg *TLSConfig) *TLSConfig {
 	}
 	if cfg.GetCertificate == nil {
 		cfg.GetCertificate = func() (tls.Certificate, error) {
-			return tls.Certificate{}, fmt.Errorf("not supported")
+			return tls.Certificate{}, MakeUnavailable(fmt.Errorf("not supported"))
```

Comment:
Don't use Unavailable for not transient error:
```
    // Unavailable indicates the service is currently unavailable.
    // This is a most likely a transient condition and may be corrected
    // by retrying with a backoff. Note that it is not always safe to retry
    // non-idempotent operations.
    //
    // See litmus test above for deciding between FailedPrecondition,
    // Aborted, and Unavailable.
    //
    // This error code will be generated by the gRPC framework during
    // abrupt shutdown of a server process or network connection.
    Unavailable Code = 14
```

Reply:
What would be a better code ?

Reply:
UNIMPLEMENTED - The operation is not implemented or is not supported/enabled in this service.